src/Illuminate/Database/Query/Builder.php

```diff
@@ -1643,43 +1646,38 @@ public function buildRawPaginator($paginator, $results, $perPage)
 	}
 
 	/**
-	 * Create a paginator for an un-grouped pagination statement.
-	 *
-	 * @param  \Illuminate\Pagination\Factory  $paginator
-	 * @param  int    $perPage
-	 * @param  array  $columns
-	 * @return \Illuminate\Pagination\Paginator
-	 */
-	protected function ungroupedPaginate($paginator, $perPage, $columns)
-	{
-		$total = $this->getPaginationCount();
-
-		// Once we have the total number of records to be paginated, we can grab the
-		// current page and the result array. Then we are ready to create a brand
-		// new Paginator instances for the results which will create the links.
-		$page = $paginator->getCurrentPage($total);
-
-		$results = $this->forPage($page, $perPage)->get($columns);
-
-		return $paginator->make($results, $total, $perPage);
-	}
-
-	/**
 	 * Get the count of the total records for pagination.
 	 *
 	 * @return int
 	 */
 	public function getPaginationCount()
 	{
-		$this->backupFieldsForCount();
+		// Do we need to use extended backup for the fields?
+		// This is only needed when there is a group by in the query
+		// Save the main query that we will need for the if we need
+		// grouping of the database query
+		$grouped = isset($this->groups) && $selectSql = $this->toSql();
 
 		// Because some database engines may throw errors if we leave the ordering
 		// statements on the query, we will "back them up" and remove them from
 		// the query. Once we have the count we will put them back onto this.
+		$this->backupFieldsForCount($grouped);
+
+		// Add the magic FROM entry when we have grouping that needs to be
+		// done by the query
+		if ($grouped)
+		{
+			$this->from = \DB::raw("({$selectSql}) as subquery");
```

Comment:
Yeah, I don't know of the underlying method of doing it, do you have a suggestion for a lower level replacement? :smiley: